vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -229,20 +234,86 @@ void vtkF3DRenderer::Initialize(const std::string& up)
 
   // Importer rely on the Environment being set, so this is needed in the initialization
   const std::regex re("([-+]?)([XYZ])", std::regex_constants::icase);
+  const std::regex re2("([+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)),"
+                       "([+-]?([0-9]+([.][0-9]*)?|[.][0-9]+)),"
+                       "([+-]?([0-9]+([.][0-9]*)?|[.][0-9]+))");
   std::smatch match;
   if (std::regex_match(up, match, re))
   {
     const float sign = match[1].str() == "-" ? -1.0 : +1.0;
     const int index = std::toupper(match[2].str()[0]) - 'X';
     assert(index >= 0 && index < 3);
 
-    this->UpIndex = index;
+    std::array<double, 3> upDir = { 0, 0, 0 };
+    upDir[index] = sign;
 
-    std::fill(this->UpVector, this->UpVector + 3, 0);
-    this->UpVector[this->UpIndex] = sign;
+    std::array<double, 3> rightDir = { 0, 0, 0 };
+    rightDir[index == 0 ? 1 : 0] = 1.0;
 
-    std::fill(this->RightVector, this->RightVector + 3, 0);
-    this->RightVector[this->UpIndex == 0 ? 1 : 0] = 1.0;
+    this->InitializeEnvironment(upDir, rightDir);
+  }
+  else if (std::regex_match(up, match, re2))
+  {
+    const std::array<double, 3> upDir = {
+      ::atof(match[1].str().c_str()), //
+      ::atof(match[4].str().c_str()), //
+      ::atof(match[7].str().c_str()), //
+    };
+    const std::array<double, 3> rightDir = { 1, 0, 0 };
+
+    this->InitializeEnvironment(upDir, rightDir);
+  }
+  else
+  {
+    F3DLog::Print(F3DLog::Severity::Warning, up + " is not a valid up direction");
+  }
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DRenderer::InitializeEnvironment(
+  const std::array<double, 3>& upDir, const std::array<double, 3>& rightDir)
+{
+  const auto isEqual = [](const std::array<double, 3>& u, const std::array<double, 3>& v)
+  {
+    constexpr double e = 1e-8;
+    return ::abs(u[0] - v[0]) < e && ::abs(u[1] - v[1]) < e && ::abs(u[2] - v[2]) < e;
+  };
+
+  /* if `up` is `(0,0,0)` make it `(0,1,0)` */
+  std::array<double, 3> up = upDir;
+  if (isEqual(up, { 0, 0, 0 }))
+  {
+    up[1] = 1.0;
+  }
+  vtkMath::Normalize(up.data());
+
+  /* make sure `right` is not `(0,0,0)` or the same as `up` */
```

Comment:
Not the same as `up` but colinear to `up`. If `right` is `p * up` for any `p` scalar, it won't work.  
It's better to check if the absolute value of the dot product between the two normalized vectors is below 0.99 or so.

Reply:
ah yes I assumed equal was enough because they'd be normalized but the can be opposite, good point!